server/src/Metrics.hs

```diff
@@ -51,9 +54,17 @@ createAndRegisterIcepeakMetrics = IcepeakMetrics
                               "Total number of bytes written to the journal so far."))
   <*> register (gauge
     (Info "icepeak_subscriber_count" "Number of websocket subscriber connections."))
+  <*> register (gauge (Info "icepeak_internal_queue_length" "Length of the internal queue."))
```

Comment:
If you make a gauge that tells you the current queue length, that tells you nothing about throughput. If Prometheus samples twice and it finds 0 both times, it could be that just nothing happened in the meantime, or it could be that 1000 items got enqueued and then dequeued in time before the next scrape. If the queue length starts growing, is it because the producer is suddenly producing more, or because the consumer stopped/slowed down consuming?

If instead you add two counters, one for the number of items added to the queue, and one for the number of items removed from the queue, then you can compute the queue length as `items_added_total - items_removed_total`, but in addition, you can compute the throughput of the producer and consumer.

You do need to be very careful to record every add/remove though, if you miss one, then `added - removed` can go out of sync with the actual queue length.